test/print_safe_buffer.c

```diff
@@ -16,8 +16,7 @@ size_t snprint_safe_buffer(char *s, size_t size, struct safe_buffer const *sb)
 {
 	size_t count = 0;
 
-	count += snprintf((s + count), (size - count), "unsigned char expected[] = {");
-	count--;
```

Comment:
I'm confused about the handling of `count` here, so had to sit down and wrap my brain around this.
Quoting the man page for `snprintf`:
> Upon successful return, these functions return the number of characters printed (excluding the null byte used to end output to strings).

So, in general, if you were to do this:
```
    count += snprintf((s + count), (size - count), "foo");
    count += snprintf((s + count), (size - count), "bar");
```
... you would produce the string `foobar`. Similarly, if you were to do this:
```
    count += snprintf((s + count), (size - count), "foo");
    count--;
    count += snprintf((s + count), (size - count), "bar");
```
... you would produce the string `fobar` (which doesn't seem to be the intent of the code).
So far, so good: the code in this PR removes the `count--`, which I agree seems to have been incorrect (and was correctly removed).
However, a few lines below (in the `for` loop), there is still a `count--;` which seems to be incorrect (it would truncate the trailing comma).
I guess I'm just confused about why the `count--;` were put there originally.

Reply:
fixed; looks like it was an error introduced in eb893c97fdbeffe3efb73de5691eaee3892efabb